internal/lsp/watcher/global_watcher.go

```diff
@@ -113,8 +115,24 @@ func Start() error {
 	events := notify.Create | notify.Write | notify.Remove | notify.Rename
 
 	if err := notify.Watch(watchPath, gw.events, events); err != nil {
+		// Check if the error might be due to file descriptor limits
+		if isFileLimitError(err) {
+			slog.Warn("lsp watcher: Hit file descriptor limit, attempting to increase", "error", err)
+			if newLimit, rlimitErr := MaximizeOpenFileLimit(); rlimitErr == nil {
+				slog.Info("lsp watcher: Increased file descriptor limit", "limit", newLimit)
+				// Retry the watch operation
+				if err = notify.Watch(watchPath, gw.events, events); err == nil {
+					slog.Info("lsp watcher: Successfully set up watch after increasing limit")
+					goto watchSuccess
+				}
+				err = fmt.Errorf("still failed after increasing limit: %w", err)
+			} else {
+				slog.Warn("lsp watcher: Failed to increase file descriptor limit", "error", rlimitErr)
+			}
```

Comment:
What about moving this block into a separate function? Something like `func attempt*() error`.

Would make code cleaner and allow to get rid of the `goto` (would become a `return`).

Reply:
Fixed the other comments, but kept this as is for now.